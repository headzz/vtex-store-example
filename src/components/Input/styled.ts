import { colors } from 'configuration'
import styled from 'styled-components'

export const Container = styled.div`
  border-radius: 2.4rem;
  position: relative;
  height: 4rem;
  width: 24rem;
`
export const InputStyled = styled.input`
  border: 1px solid ${colors.GREY_LVL1};
  border-radius: 2.4rem;
  color: ${colors.PINK_JUSSI};
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem 0.8rem 1.6rem;
  outline: 0;
  width: 100%;

  ::placeholder {
    color: ${colors.PINK_JUSSI};
    opacity: 1;
  }
`
export const ButtonContainer = styled.div`
  padding-right: 1.6rem;
  position: absolute;
  right: 0;
  top: 0.8rem;
`
