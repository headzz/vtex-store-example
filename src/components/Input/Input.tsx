import Button from 'components/Button'
import { useState } from 'react'
import { ButtonContainer, Container, InputStyled } from './styled'

/**
 * @component
 * @name Input
 * @description creates Input component to use on screen.
 * @namespace components components/Input
 *
 */
const Input: React.FC = () => {
  const [value, setValue] = useState('')
  return (
    <Container>
      <InputStyled
        type="text"
        placeholder="Buscar"
        onChange={(event) => setValue(event.target.value)}
        value={value}
      />
      <ButtonContainer>
        <Button
          icon="search"
          color=""
          round={false}
          clickHandler={() => console.log(value)}
        />
      </ButtonContainer>
    </Container>
  )
}

export default Input
