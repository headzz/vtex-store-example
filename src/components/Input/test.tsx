import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import Input from './Input'

describe('Input', () => {
  it('should render component input', () => {
    const { container } = render(<Input />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render text after typing', () => {
    render(<Input />)

    userEvent.type(screen.getByPlaceholderText('Buscar'), 'Hello, World!')
    expect(screen.getByRole('textbox')).toHaveValue('Hello, World!')

    expect(screen.getByRole('button')).toBeInTheDocument()
  })
})
