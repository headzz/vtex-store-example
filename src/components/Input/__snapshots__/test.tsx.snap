// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render component input 1`] = `
.c3 {
  border: 0;
  border-radius: 0;
  background-color: transparent;
  outline: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0 {
  border-radius: 2.4rem;
  position: relative;
  height: 4rem;
  width: 24rem;
}

.c1 {
  border: 1px solid #F2F2F2;
  border-radius: 2.4rem;
  color: #EE0E7D;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem 0.8rem 1.6rem;
  outline: 0;
  width: 100%;
}

.c1::-webkit-input-placeholder {
  color: #EE0E7D;
  opacity: 1;
}

.c1::-moz-placeholder {
  color: #EE0E7D;
  opacity: 1;
}

.c1:-ms-input-placeholder {
  color: #EE0E7D;
  opacity: 1;
}

.c1::placeholder {
  color: #EE0E7D;
  opacity: 1;
}

.c2 {
  padding-right: 1.6rem;
  position: absolute;
  right: 0;
  top: 0.8rem;
}

<div
  class="c0"
>
  <input
    class="c1"
    placeholder="Buscar"
    type="text"
    value=""
  />
  <div
    class="c2"
  >
    <button
      class="c3"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <img
          alt="Ã­cone de pesquisa"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </button>
  </div>
</div>
`
